torchaudio/models/deepspeech.py

```diff
@@ -0,0 +1,88 @@
+import torch
+import torch.nn as nn
+
+__all__ = ["DeepSpeech"]
+
+
+class FullyConnected(nn.Module):
+    """
+    Args:
+        in_features: Number of input features
+        hidden_size: Internal hidden unit size.
+    """
+
+    def __init__(self,
+                 in_features: int,
+                 hidden_size: int,
+                 dropout: float,
+                 relu_max_clip: int = 20) -> None:
+        super(FullyConnected, self).__init__()
+        self.fc = nn.Linear(in_features, hidden_size, bias=True)
+        self.nonlinearity = nn.Sequential(*[
+            nn.ReLU(),
+            nn.Hardtanh(0, relu_max_clip)
+        ])
+        if dropout:
+            self.nonlinearity = nn.Sequential(*[
+                self.nonlinearity,
+                nn.Dropout(dropout)
+            ])
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = self.fc(x)
+        x = self.nonlinearity(x)
+        return x
+
+
+class DeepSpeech(nn.Module):
+    """
+    DeepSpeech model architecture from
+    `"Deep Speech: Scaling up end-to-end speech recognition"`
+    <https://arxiv.org/abs/1412.5567> paper.
+
+    Args:
+        in_features: Number of input features
+        hidden_size: Internal hidden unit size.
+        num_classes: Number of output classes
+    """
+
+    def __init__(self,
+                 in_features: int,
+                 hidden_size: int = 2048,
+                 num_classes: int = 40,
+                 dropout: float = 0.0) -> None:
+        super(DeepSpeech, self).__init__()
+        self.hidden_size = hidden_size
+        self.fc1 = FullyConnected(in_features, hidden_size, dropout)
+        self.fc2 = FullyConnected(hidden_size, hidden_size, dropout)
+        self.fc3 = FullyConnected(hidden_size, hidden_size, dropout)
+        self.bi_rnn = nn.RNN(
+            hidden_size, hidden_size, num_layers=1, nonlinearity='relu', bidirectional=True)
+        self.nonlinearity = nn.ReLU()
+        self.fc4 = FullyConnected(hidden_size, hidden_size, dropout)
+        self.out = nn.Sequential(*[
+            nn.Linear(hidden_size, num_classes),
+            nn.LogSoftmax(dim=2)
+        ])
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
```

Comment:
Can you update the forward pass with the shape of the tensors? You can see the docstrings for the other models [here](https://github.com/pytorch/audio/tree/master/torchaudio/models). 

Also, note that we've been putting batch first (which is compatible with `DataParallel` also) in torchaudio. [If you like the flexibility, we could have a flag `batch_first` default to `True` also (though I know that [rnn](https://pytorch.org/docs/stable/generated/torch.nn.RNN.html) default to `False` unfortunately).]

Reply:
> Also, note that we've been putting batch first (which is compatible with DataParallel also) in torchaudio. [If you like the flexibility, we could have a flag batch_first default to True also (though I know that rnn default to False unfortunately).]

Apologies. I did not get this part of comment. Do you mean to put `batch_size` first in the model output?

Reply:
no worries, let me clarify what I meant there :) 

if you look at [wav2letter](https://github.com/pytorch/audio/blob/master/torchaudio/models/wav2letter.py#L67), [convtasnet](https://github.com/pytorch/audio/blob/master/torchaudio/models/conv_tasnet.py#L297), [wavernn](https://github.com/pytorch/audio/blob/master/torchaudio/models/wavernn.py#L280), all take as input a tensor where the first dimension is a batch. we need to make the model consume a tensor with batch first like the others.

the reference to `batch_first` was made in connection to the interface for [rnn in pytorch](https://pytorch.org/docs/stable/generated/torch.nn.RNN.html#torch.nn.RNN). however, let me change my mind on this: let's not add a `batch_first` parameter to the constructor since no other model have that in torchaudio.

Reply:
> if you look at wav2letter, convtasnet, wavernn, all take as input a tensor where the first dimension is a batch. we need to make the model consume a tensor with batch first like the others.

ok, it's clear now. I'll add the similar docstrings.